### PR TITLE
[1.21.1] Update Entity Icons

### PR DESCRIPTION
**Pull request by UnRealDinnerbone**
- Rework how entity icons are loaded
- Allow icons to be load from the mob texture
- Make mob icons configurable to if they display or not
- Add icons for missing mobs

There a few files that have there formatting changed (Seems to be tabs to spaces)  Don't know what to do about those.

Should I just leave them or make my formatter match better?

### CODE REVIEW

File: common/src/main/java/dev/ftb/mods/ftbchunks/client/FTBChunksClient.java [hunk 1]
```diff
@@ -94,1187 +133,1202 @@
...
+        } else {
+            zoom += zoomFactor;
+        }
+
+        lastZoomTime = System.currentTimeMillis();
+        FTBChunksClientConfig.MINIMAP_ZOOM.set(zoom);
+        return EventResult.interruptTrue();
+    }
+
+    private static long clamp(long val, long min, long max) {
```
Comment (MichaelHillcox) [inline common/src/main/java/dev/ftb/mods/ftbchunks/client/FTBChunksClient.java:482]: Likely not a you issue as this code has just been reformatted but this should just use the `Mth` class

File: common/src/main/java/dev/ftb/mods/ftbchunks/client/gui/EntityIconSettingsScreen.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        @Override
+        public void alignWidgets() {
+
+        }
+
+        @Override
+        public void setWidth(int newWidth) {
+            super.setWidth(newWidth);
+
+            if (newWidth > 0) {
```
Comment (MichaelHillcox) [inline common/src/main/java/dev/ftb/mods/ftbchunks/client/gui/EntityIconSettingsScreen.java:111]: Return early

File: common/src/main/java/dev/ftb/mods/ftbchunks/client/gui/SliceCreationGUI.java [hunk 3]
```diff
@@ -0,0 +1,362 @@
+package dev.ftb.mods.ftbchunks.client.gui;
```
Comment (MichaelHillcox) [inline common/src/main/java/dev/ftb/mods/ftbchunks/client/gui/SliceCreationGUI.java:1]: This file is pretty messy, please clean up the code and add some comments here and there to explain the more weird sections

File: common/src/main/java/dev/ftb/mods/ftbchunks/client/mapicon/EntityImageIcon.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+    }
+
+    public record Slice(
+            int x,
+            int y,
+            int width,
+            int height) {
+
+        public static final Codec<Slice> CODEC = RecordCodecBuilder.create(builder -> builder.group(
```
Comment (MichaelHillcox) [inline common/src/main/java/dev/ftb/mods/ftbchunks/client/mapicon/EntityImageIcon.java:77]: Can't you combine codecs to create a combination of the offset and width and height ones?

**Review by MichaelHillcox — APPROVED** [review]
Only a few notes. Please avoid code reformatting in the future unless it's in a dedicated PR

**Review by desht — APPROVED** [review]

**Pull request merged**